File: src/main/java/cloud/common/Suspect.java
package cloud.common;

import se.sics.kompics.Event;
import se.sics.kompics.address.Address;

/**
 * 
 * @author Amir Moulavi
 * @date 2011-03-27
 *
 */

public class Suspect extends Event {

	private Address node;

	public Suspect(Address node) {
		this.node = node;
	}

	public Address getNode() {
		return node;
	}

}

File: src/main/java/cloud/common/Suspect.scala
package cloud.common

import se.sics.kompics.address.Address
import se.sics.kompics.Event
import reflect.BeanProperty

/**
 * @author Amir Moulavi
 * @date 2011-07-09
 *
 **/
class Suspect(@BeanProperty var node:Address) extends Event